validator/client/src/test/java/tech/pegasys/teku/validator/client/RuntimeProposerConfigTest.java

```diff
@@ -36,42 +37,69 @@ class RuntimeProposerConfigTest {
   private final BLSPublicKey pubkey2 = dataStructureUtil.randomPublicKey();
   private final Eth1Address address = dataStructureUtil.randomEth1Address();
   private final Eth1Address address2 = dataStructureUtil.randomEth1Address();
+  private final UInt64 gasLimit = dataStructureUtil.randomUInt64();
+  private final UInt64 gasLimit2 = dataStructureUtil.randomUInt64();
 
   @Test
-  void shouldAddEntries() {
+  void shouldAddFeeRecipientEntries() {
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).isEmpty();
-    proposerConfig.addOrUpdate(pubkey, address);
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
   }
 
   @Test
-  void shouldUpdateEntries() {
-    proposerConfig.addOrUpdate(pubkey, address);
+  void shouldAddGasLimitEntries() {
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).isEmpty();
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+  }
+
+  @Test
+  void shouldUpdateFeeRecipientEntries() {
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
-    proposerConfig.addOrUpdate(pubkey, address2);
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address2);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address2);
   }
 
   @Test
-  void shouldDeleteEntries() {
-    proposerConfig.addOrUpdate(pubkey, address);
+  void shouldUpdateGasLimitEntries() {
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit2);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit2);
+  }
+
+  @Test
+  void shouldDeleteFeeRecipientEntries() {
+    proposerConfig.addOrUpdateFeeRecipient(pubkey, address);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).contains(address);
-    proposerConfig.delete(pubkey);
+    proposerConfig.deleteFeeRecipient(pubkey);
     assertThat(proposerConfig.getEth1AddressForPubKey(pubkey)).isEmpty();
   }
 
+  @Test
+  void shouldDeleteGasLimitEntries() {
+    proposerConfig.addOrUpdateGasLimit(pubkey, gasLimit);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).contains(gasLimit);
+    proposerConfig.deleteGasLimit(pubkey);
+    assertThat(proposerConfig.getGasLimitForPubKey(pubkey)).isEmpty();
+  }
```

Comment:
this looks good, similar theory, call delete on empty and see it doesn't do anything silly...